modules/installing-oadp-aws-sts.adoc

```diff
@@ -0,0 +1,260 @@
+gp3-csi (default)   ebs.csi.aws.com         Delete          WaitForFirstConsumer   true                   4d21h
+----
++
+[NOTE]
+====
+The following storage classes will work:
+
+  * gp3-csi
+  * gp2-csi
+  * gp3
+  * gp2
+====
++
+If the application or applications that are being backed up are all using persistent volumes (PVs) with Container Storage Interface (CSI), it is advisable to include the CSI plugin in the OADP DPA configuration.
+
+. Create the `DataProtectionApplication` resource to configure the connection to the storage where the backups and volume snapshots are stored:
+
+.. If you are using only CSI volumes, deploy a Data Protection Application by entering the following command:
++
+[source,terminal]
+----
+$ cat << EOF | oc create -f -
+  apiVersion: oadp.openshift.io/v1alpha1
+  kind: DataProtectionApplication
+  metadata:
+    name: ${CLUSTER_NAME}-dpa
+    namespace: openshift-adp
+  spec:
+    backupImages: true <1>
+    features:
+      dataMover:
+        enable: false
+    backupLocations:
+    - bucket:
+        cloudStorageRef:
+          name: ${CLUSTER_NAME}-oadp
+        credential:
+          key: credentials
+          name: cloud-credentials
+        prefix: velero
+        default: true
+        config:
+          region: ${REGION}
+    configuration:
+      velero:
+        defaultPlugins:
+        - openshift
+        - aws
+        - csi
+      restic:
+        enable: false
+EOF
+----
+<1> Set this field to `false` if you do not want to use image backup.
+
+// . Create the `DataProtectionApplication` resource, which is used to configure the connection to the storage where the backups and volume snapshots are stored:
+
+.. If you are using CSI or non-CSI volumes, deploy a Data Protection Application by entering the following command:
++
+[source,terminal]
+----
+$ cat << EOF | oc create -f -
+  apiVersion: oadp.openshift.io/v1alpha1
+  kind: DataProtectionApplication
+  metadata:
+    name: ${CLUSTER_NAME}-dpa
+    namespace: openshift-adp
+  spec:
+    backupImages: true <1>
+    features:
+      dataMover:
+         enable: false
+    backupLocations:
+    - bucket:
+        cloudStorageRef:
+          name: ${CLUSTER_NAME}-oadp
+        credential:
+          key: credentials
+          name: cloud-credentials
+        prefix: velero
+        default: true
+        config:
+          region: ${REGION}
+    configuration:
+      velero:
+        defaultPlugins:
+        - openshift
+        - aws
+      nodeAgent: <2>
+        enable: false
+        uploaderType: restic
+    snapshotLocations:
+      - velero:
+          config:
+            credentialsFile: /tmp/credentials/openshift-adp/cloud-credentials-credentials <3>
+            enableSharedConfig: "true" <4>
+            profile: default <5>
+            region: ${REGION} <6>
+          provider: aws
+EOF
+----
+<1> Set this field to false if you do not want to use image backup.
+<2> See the following note.
```

Comment:
There are 2 NOTES. Does this apply to the first NOTE? 

I wonder if something like this might be more clear:

.. If you are using CSI or non-CSI volumes, deploy a Data Protection Application by entering the following command:
+
--
[source,terminal]
----
$ cat << EOF | oc create -f -
  apiVersion: oadp.openshift.io/v1alpha1
  kind: DataProtectionApplication
  metadata:
    name: ${CLUSTER_NAME}-dpa
    namespace: openshift-adp
  spec:
    backupImages: true <1>
    features:
      dataMover:
         enable: false
    backupLocations:
    - bucket:
        cloudStorageRef:
          name: ${CLUSTER_NAME}-oadp
        credential:
          key: credentials
          name: cloud-credentials
        prefix: velero
        default: true
        config:
          region: ${REGION}
    configuration:
      velero:
        defaultPlugins:
        - openshift
        - aws
      nodeAgent: <2>
        enable: false
        uploaderType: restic
    snapshotLocations:
      - velero:
          config:
            credentialsFile: /tmp/credentials/openshift-adp/cloud-credentials-credentials <3>
            enableSharedConfig: "true" <4>
            profile: default <5>
            region: ${REGION} <6>
          provider: aws
EOF
----
<1> Set this field to false if you do not want to use image backup.
<2> The node agent configuration (or whatever this is).
+
[NOTE]
====
If you use OADP 1.2, replace this configuration:
[source,terminal]
----
nodeAgent:
  enable: false
  uploaderType: restic
----
with the following configuration:
[source,terminal]
----
restic:
  enable: false
----
====
<3> The `credentialsFile` field is the mounted location of the bucket credential on the pod.
<4> The `enableSharedConfig` field allows the `snapshotLocations` to share or reuse the credential defined for the bucket.
<5> Use the profile name set in the {aws-short} credentials file.
<6> Specify `region` as your {aws-short} region. This must be the same as the cluster region.
--
+
You are now ready to back up and restore {product-title} applications, as described in _Backing up applications_.
+
[NOTE]
====
If you want to use two different clusters for backing up and restoring, the two clusters must have the same {aws-short} S3 storage names in both the cloud storage CR and the OADP `DataProtectionApplication` configuration.
====

It would also more closely align with admonition style https://redhat-documentation.github.io/supplementary-style-guide/#admonitions "Keep admonitions to a minimum, and avoid placing multiple admonitions close to one another." 

However, I'm not sure if adding a NOTE inside a callout will work.

Another option is to make it a separate step, like Step B.

Reply:
Wanted to add a comment to mention that you may need to use open blocks: (https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/#enclose-in-open-block), and that someone else asked about an admonition in a call out, and Kathryn asked with this i as an example: https://github.com/openshift/openshift-docs/blame/main/modules/installation-gcp-config-yaml.adoc#L197

Reply:
Also just learned there's a PR in flight that provides guidance on how to put an admonition in a callout: https://github.com/openshift/openshift-docs/pull/72401/files roughly lines 50-70. Hope it helps!

Reply:
@gwynnemonahan 
Thank you for raising this issue.
Does this work?
<2> See the first note regarding the `nodeAgent` attribute.

Reply:
Even with clarifying which Note, the admonitions are still stacked on top of each other, which is not in line with style: admonition style https://redhat-documentation.github.io/supplementary-style-guide/#admonitions

Reply:
@gwynnemonahan 

This exact formatting was previously approved and already published last month in parallel documentation.

https://access.redhat.com/documentation/en-us/openshift_container_platform/4.15/html/backup_and_restore/oadp-application-backup-and-restore#oadp-rosa-backing-up-applications

Can we please leave this as it is?

Reply:
OK. Then you'll want to create a Jira issue for this, and other instances of stacked admonitions, so you can go back and bring them in line with current style at a later date.

Reply:
@gwynnemonahan 
Will do. Thank you!!!

Reply:
https://issues.redhat.com/browse/OADP-3768